src/Microsoft.AspNetCore.Blazor/Components/Provider.cs

```diff
@@ -85,6 +85,17 @@ public void SetParameters(ParameterCollection parameters)
             _renderHandle.Render(Render);
         }
 
+        internal override bool CanSupplyValue(Type requestedType, string requestedName)
+        {
+            if (!requestedType.IsAssignableFrom(typeof(T)))
+            {
+                return false;
+            }
+
+            return (requestedName == null && Name == null) // Match on type alone
+                || string.Equals(requestedName, Name, StringComparison.Ordinal); // Also match on name
```

Comment:
Are we sure about this being case sensitive when things like attribute/parameter names are case-insensitive?

Reply:
Honestly I'm not sure about it. My real preference would be for attribute/parameter names to be case sensitive too, but we went with insensitive because tag helpers are insensitive, which of course goes back to the history with it being a DSL for HTML (thanks, Berners-Lee!).

In this case I picked sensitive just because it's safer to be stricter, in that we can (more) safely loosen it in the future. But TBH you're right that the inconsistency is awkward. I'll change it to be case-insensitive.

Reply:
Changed to case-insensitive.

Reply:
Yeah - I personally prefer that all identifiers in programming systems I use are case-sensitive. I feel like case-insensitive is something we only do at microsoft and I've never understood why it's valuable.